monitoring/metrics.go

```diff
@@ -0,0 +1,125 @@
+/*
+Copyright 2017 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package monitoring
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"time"
+
+	"github.com/gravitational/trace"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+const collectETCDMetricsTimeout = 5 * time.Second
+
+// EtcdLeaderStats is used by the leader in an etcd cluster, and encapsulates
+// statistics about communication with its followers
+type EtcdLeaderStats struct {
+	// Leader is the ID of the leader in the etcd cluster.
+	Leader    string                        `json:"leader"`
+	Followers map[string]*EtcdFollowerStats `json:"followers"`
```

Comment:
You should document all exported fields.
Also, can you link to the etcd metric schema documentation/reference?

`EtcdLeaderStats encapsulates leader/followers metrics in an etcd cluster`